### DISCUSSION TITLE
How to expose the full classpath from classloader?

### DISCUSSION THREAD

**Discussion by cowwoc** [classgraph/classgraph#820] [General]
Could you please clarify what this sentence from the README.md file means?

> Therefore, please convince your upstream runtime environment maintainers to expose the full classpath from their classloader using a public method or field, otherwise ClassGraph may stop working for your runtime environment in the future.

It's not clear to me what "runtime environment maintainers" refers to.

Are you asking ClassLoader implementors (including JVM vendors) to provide public methods to look up the classpath? Or are you asking individual library authors to do something?

If you are talking about ClassLoader implementors, what about the standard/built-in ClassLoader implementations that ship with Java? Do they already provide a public method to do this? Or do we need to ask Oracle to add such a method?

**Comment by lukehutch** [comment]
No, I'm talking about whoever wrote the custom classloader in whatever environment you are running in (eg. JBoss, Spring, whatever runs your classes). Many classloader implementations don't have a public API for reading the class path that the classloader is using, because there is no public method for that that has to be implemented in the `ClassLoader` API.

**Reply by cowwoc** [reply]
Got it. Thank you for the clarification.

So what happens for the classloaders in the core Java API? How are you able to look up the classpath for them?
